### PR TITLE
Fix before beta

### PR DESCRIPTION
**Pull request by LaurentPetard**

### CODE REVIEW

File: src/Client/HttpExceptionHandler.php [hunk 1]
```diff
@@ -73,7 +73,10 @@ public function transformResponseToException(RequestInterface $request, Response
      */
     protected function getResponseMessage(ResponseInterface $response)
     {
-        $decodedBody = json_decode($response->getBody()->getContents(), true);
+        $responseBody = $response->getBody();
+        $decodedBody = json_decode($responseBody->getContents(), true);
+
+        $responseBody->rewind();
```
Comment (skeleton) [inline src/Client/HttpExceptionHandler.php]: can you explain ? it's a little weird

Comment (LaurentPetard) [inline src/Client/HttpExceptionHandler.php]: The response body is a Stream. When it's read by calling getContents, the internal cursor stay at the end of the stream. So then getContents is called a second time it returns an empty string. the function rewind put the internal cursor at the beginning of the stream.

Comment (ahocquard) [inline src/Client/HttpExceptionHandler.php]: You should do it before reading the message, not after. Prevent the bug before reading, not after.

File: src/Exception/UnprocessableEntityHttpException.php [hunk 2]
```diff
@@ -18,7 +18,10 @@ class UnprocessableEntityHttpException extends ClientErrorHttpException
      */
     public function getResponseErrors()
     {
-        $decodedBody = json_decode($this->getResponse()->getBody()->getContents(), true);
+        $responseBody = $this->response->getBody();
```
Comment (ahocquard) [inline src/Exception/UnprocessableEntityHttpException.php]: same

File: src/Client/HttpExceptionHandler.php (hunk 1 above)
Comment (LaurentPetard) [inline src/Client/HttpExceptionHandler.php]: So it should be done before AND after. The API users can get the response from the exception, and  they surely don't think to do a rewind if they want to read the body.

**Pull request merged**